warp10/src/main/java/io/warp10/WarpRun.java

```diff
@@ -25,61 +25,97 @@
 import io.warp10.continuum.Configuration;
 import io.warp10.script.MemoryWarpScriptStack;
 import io.warp10.script.StackUtils;
+import io.warp10.script.WarpScriptLib;
+import io.warp10.script.ext.debug.DebugWarpScriptExtension;
+import io.warp10.script.ext.warprun.WarpRunWarpScriptExtension;
 import io.warp10.script.functions.SNAPSHOT;
 
 public class WarpRun {
-  
+
   private static final String WARPRUN_FORMAT = "warprun.format";
-  
+
   public static void main(String[] args) throws Exception {
     try {
       System.setProperty(Configuration.WARP10_QUIET, "true");
       System.setProperty(Configuration.WARPSCRIPT_REXEC_ENABLE, "true");
-      
+
       if (null == System.getProperty(Configuration.WARP_TIME_UNITS)) {
         System.setProperty(Configuration.WARP_TIME_UNITS, "us");
       }
-      
-      WarpConfig.setProperties((String) null);
-      
+
+      //
+      // Initialize WarpConfig
+      //
+
+      try {
+        String config = System.getProperty(WarpConfig.WARP10_CONFIG);
+
+        if (null == config) {
+          config = System.getenv(WarpConfig.WARP10_CONFIG_ENV);
+        }
+
+        String[] files = null;
+
+        if (null != config) {
+          files = config.split("[, ]+");
+        }
+
+        WarpConfig.safeSetProperties(files);
+        WarpScriptLib.registerExtensions();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+
       Properties properties = WarpConfig.getProperties();
-      
+
+      //
+      // Register the WarpRun and Debug extensions
+      //
+
+      WarpScriptLib.register(new WarpRunWarpScriptExtension());
+      WarpScriptLib.register(new DebugWarpScriptExtension());
+
       MemoryWarpScriptStack stack = new MemoryWarpScriptStack(null, null, properties);
       stack.maxLimits();
-      
+
```

Comment:
Break condition should be change at line 94

Reply:
?

Reply:
When dealing with a "-" parameter, the reading process never end.
The `if (null == line) {` should be adapted.

Reply:
When stdin reaches EOF, readline will return null, do you have a test case where that is not the case?

Reply:
No, I misused it